Create a GenreLabelView as a common component

vaib-ios opened the pull request:
Fixed some errors from previous PR and added a optional chaining 
Added a new view to show Genre and Rating View

vaib-ios committed:
Create a GenreLabelView as a common component

Fixed some errors from previous PR and added a optional chaining
Added a new view to show Genre and Rating View

kkpratik commented on MovieShowcase/Common/GenreLabelView.swift:
```diff
@@ -0,0 +1,43 @@
+                .scaledToFit()
+                .frame(height: 13)
+                .foregroundStyle(.yellow)
+                
```
Could we remove the extra space.

kkpratik commented on MovieShowcase/Common/GenreLabelView.swift:
```diff
@@ -0,0 +1,43 @@
+                                .stroke(Color.gray.opacity(0.5))
+                        )
+                )
+        
```
Could we remove the extra space.

vaib-ios committed:
Address the feeback

- Added missing optional chaining in carouselview for title variable
- Remove unwanted extra lines
- Remove boiler plate headers from the top of the GenreLabelView

kkpratik commented:
Looks good.
